api/src/main/java/io/strimzi/api/kafka/model/KafkaClusterRebalanceSpec.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.dataformat.yaml.YAMLMapper;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({ "goals", "verbose" })
+@EqualsAndHashCode
+public class KafkaClusterRebalanceSpec implements UnknownPropertyPreserving, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private List<String> goals;
+    private boolean verbose = false;
+
+    private Map<String, Object> additionalProperties = new HashMap<>(0);
+
+    @Description("A list of goals to use in generating the proposal and executing it")
+    public List<String> getGoals() {
+        return goals;
```

Comment:
How are the goals defined?

Reply:
@scholzj see conversation [here](https://github.com/strimzi/strimzi-kafka-operator/pull/2306/files#r356703289).

@ppatierno I think we need a better `@Description` here.

Reply:
@tombentley what do you have in mind? Listing all the possible goals already available on CC (they are almost 20 or so).

Reply:
Well, you could mention a few choice goals and/or link to some documentation which discusses goals in more detail (that could be the CC upstream doc for now, and maybe some Strimzi docs in the future).